contracts/Elections.sol

```diff
@@ -57,68 +47,75 @@ contract Elections is IElections, IStakeChangeNotifier, Ownable {
-		maxTopologySize = _maxTopologySize;
+		minCommitteeSize = _minCommitteeSize;
 	    maxDelegationRatio = _maxDelegationRatio;
 		voteOutPercentageThreshold = _voteOutPercentageThreshold;
 		voteOutTimeoutSeconds = _voteOutTimeoutSeconds;
 		banningPercentageThreshold = _banningPercentageThreshold;
 	}
 
-	function setContractRegistry(IContractRegistry _contractRegistry) external onlyOwner {
-		require(_contractRegistry != IContractRegistry(0), "contractRegistry must not be 0");
-		contractRegistry = _contractRegistry;
-	}
-
-	function getTopology() external view returns (address[] memory) {
-		return topology;
-	}
-
 	/// @dev Called by: validator registration contract
 	/// Notifies a new validator was registered
 	function validatorRegistered(address addr) external onlyValidatorsRegistrationContract {
-		_rankValidator(addr);
+		if (_isBanned(addr)) {
+			return;
+		}
+		addMemberToCommittees(addr);
 	}
 
 	/// @dev Called by: validator registration contract
 	/// Notifies a new validator was unregistered
 	function validatorUnregistered(address addr) external onlyValidatorsRegistrationContract {
-		_rankValidator(addr);
+		removeMemberFromCommittees(addr);
 	}
 
 	/// @dev Called by: validator registration contract
-	/// Notifies a validator's IP has been changed
-	function validatorIpChanged(address addr) external  onlyValidatorsRegistrationContract {
-		(, bool isInTopology) = _findInTopology(addr);
-		if (isInTopology) {
-			_notifyTopologyChanged();
+	/// Notifies on a validator compliance change
+	function validatorConformanceChanged(address addr, string calldata conformanceType) external onlyComplianceContract {
+		if (_isBanned(addr)) {
+			return;
+		}
+
+		if (isComplianceType(conformanceType)) {
+			complianceCommitteeContract().addMember(addr, getCommitteeEffectiveStake(addr));
+		} else {
+			complianceCommitteeContract().removeMember(addr);
 		}
 	}
 
-	/// @dev Called by: validator registration contract
-	/// Notifies a validator's Orbs address has been changed
-	function validatorOrbsAddressChanged(address addr) external  onlyValidatorsRegistrationContract {
-		(uint pos, bool isInTopology) = _findInTopology(addr);
-		if (isInTopology) {
-			_notifyTopologyChanged();
-			if (pos < committeeSize) {
-				_notifyCommitteeChanged();
-			}
+	function notifyReadyForCommittee() external onlyNotBanned {
+		address sender = getMainAddrFromOrbsAddr(msg.sender);
+		(bool committeeChanged,) = generalCommitteeContract().memberReadyForCommittee(sender);
+		if (committeeChanged) {
+			updateComplianceCommitteeMinimumWeight();
 		}
+		complianceCommitteeContract().memberReadyForCommittee(sender);
 	}
 
-	function notifyReadyForCommittee() external {
+	function notifyReadyToSync() external onlyNotBanned {
 		address sender = getMainAddrFromOrbsAddr(msg.sender);
-		readyValidators[sender] = true;
-		_rankValidator(sender);
+		(bool committeeChanged,) = generalCommitteeContract().memberReadyToSync(sender);
```

Comment:
I like the fact that the elections contract is agnostic to these technicalities - if we upgrade the committee contract and change this behaviour the election contract could stay the same.

Reply:
I agree that as long as the `Committee` returns the flag `Elections` must check it. but i'm not sure why `Committee` returns this flag at all but since its "spec" we can postpone this to a later review